src/app.py

```diff
@@ -2001,6 +2243,220 @@ def upload_data_status():
+sub_type : str
+        The sample category of the given samples
+validate_uuids: int
+    An integer value 1 or 0 indicating whether the individual samples should be validated along with the schema. 
+Returns
+--------
+message : json array of representing the individual errors returned from ingest-validation-tools
+200: json containing {'message': 'success'}
+"""
+@app.route('/sample-bulk-metadata', methods=['PUT'])
+def sample_bulk_metadata():
+    if 'file' not in request.files:
+        bad_request_error('No file part')
+    file = request.files['file']
+    sub_type = request.form.get('sub_type')
+    validate_uuids = request.form.get('validate_uuids')
+    if sub_type is None:
+        bad_request_error('No sub_type in request')
+    if validate_uuids is None:
+        validate_uuids = "0"
+    data = {
+        "entity_type": "sample",
+        "sub_type": sub_type,
+        "validate_uuids": validate_uuids
+    }
+    if file.filename == '':
+        bad_request_error('No selected file')
+    file.filename = file.filename.replace(" ", "_")
+    token = auth_helper_instance.getAuthorizationTokens(request.headers)
+    header = {'Authorization': 'Bearer ' + token}
+    try:
+        temp_id = file_upload_helper_instance.save_temp_file(file)
+    except Exception as e:
+        bad_request_error(f"Failed to create temp_id: {e}")
+    file.filename = utils.secure_filename(file.filename)
+    path_name = temp_id + os.sep + file.filename
+    file_details = {
+        "filename": os.path.basename(path_name),
+        "pathname": path_name,
+        "fullpath": commons_file_helper.ensureTrailingSlash(app.config['FILE_UPLOAD_TEMP_DIR']) + temp_id + os.sep + file.filename
+    }
+    message = validate_uploaded_metadata(file_details, token, data)
+    if len(message) > 0:
+        return jsonify(f"Errors occurred during validation. {message}"), 400
+    headers = []
+    records = []
+    with open(file_details['fullpath'], newline="") as tsvfile:
+        reader = csv.DictReader(tsvfile, delimiter="\t")
+        first = True
+        for row in reader:
+            data_row = {}
+            for key in row.keys():
+                if first:
+                    headers.append(key)
+                data_row[key] = row[key]
+            records.append(data_row)
+            if first:
+                first = False
+    updates = []
+    ids = []
+    for r in records:
+        id = r.get('sample_id')
+        update = {
+            'id': id,
+            'metadata': json.dumps(r).replace('"', "'") 
+        }
+        updates.append(update)
+        ids.append(id)
+
+    update_query = "WITH ["
+    update_query += ", ".join([f"{{id: \"{u['id']}\", metadata: \"{u['metadata']}\"}}" for u in updates])
+    update_query += f"] AS updates UNWIND updates AS u MATCH (e {{hubmap_id: u.id}}) SET e.metadata = u.metadata"
+    try:
+        with neo4j_driver_instance.session() as neo_session:
+            result = neo_session.run(update_query).data()
```

Comment:
Because we are updating the sample metadata, this Neo4j query should be handled within a Transaction session. If the transaction fails to commit by any reason, an explicit rollback should happen.